src/renderer/src/components/Sidebar/SettingsSidebarLink.tsx

```diff
@@ -0,0 +1,32 @@
+import { cloneElement, useEffect, useState } from 'react'
+
+export const SettingsSidebarLink = ({ id, icon, title, onClick, selectedId }): JSX.Element => {
```

Comment:
Create a type for the props
```suggestion
type TProps = {
icon: JSX.Element
title: string
onClick: MouseEventHandler<HTMLButtonElement>
}

export const SettingsSidebarLink = ({ id, icon, title, onClick, selectedId }): JSX.Element => {
...component
}
```